### ISSUE TITLE
Horizontal line plot fails on y = 1.0000000000000009

### ISSUE THREAD

**Issue by RFIsoft** [swharden/ScottPlot#607]
Hello,
Great software!

On a line plot, with a horizontal line, a y value of 1.0000000000000009, the plot will hang on setting the title (plt.Title("myTitle").
When rounding the y, it works fine. That's the work around I use now.
In the attached example, the x runs from about 50 to 150.

Tx, Pete

![Screenshot 2020-10-26 135908](https://user-images.githubusercontent.com/32685182/97228834-b093b180-1794-11eb-8e47-b57cd4e54436.png)

**Comment by swharden** [comment]
Hi @RFIsoft, thanks for reporting this! I'm glad you are enjoying ScottPlot.

I'm trying to recreate your bug in a simpler form:

```cs
formsPlot1.plt.PlotLine(x1: 0, y1: 1, x2: 2, y2: 3);
formsPlot1.Render();
```
![image](https://user-images.githubusercontent.com/4165489/97232082-eb0c4700-17b2-11eb-816c-239b966472e5.png)


```cs
formsPlot1.plt.PlotLine(x1: 0, y1: 1, x2: 2, y2: 1.0000000000000009);
```
![image](https://user-images.githubusercontent.com/4165489/97232178-14c56e00-17b3-11eb-9179-49c46092ad1f.png)

I think the problem is that the plot is that auto-axis is failing since the Y deviation is so low. I'll work to make this better, but for now I think you can fix this by manually defining an axis after plotting.

```cs
formsPlot1.plt.Axis(y1: 0, y2: 2);
```

**Comment by RFIsoft** [comment]
Thanks Scott... Note that I had a horizontal line (so formsPlot1.plt.PlotLine(x1: 0, y1: 1.0000000000000009, x2: 2, y2: 1.0000000000000009);
With your observation, I should have mentioned: I did plot points as well before this. And these where all on one line, y=1. So your observation might be correct.

I recreated a scenario in LinqPad6, this will hang and not finish:
```
var filename = @"C:\temp\sp.png";
var plt = new ScottPlot.Plot(600, 400);
var y  = 1.0000000000000009;
for (var i = 0; i < 10; i++)
{
    plt.PlotPoint(i,1);
}
plt.PlotLine(x1: 0, y1: y, x2: 9, y2: y);
plt.SaveFig(filename);
ProcessStartInfo psi = new ProcessStartInfo(filename) { UseShellExecute = true };
Process.Start(psi);
```

I realize it's really a cornercase I did run into. But it might point to something to enhance internally.

Thanks again for this gem!

**Comment by swharden** [comment]
This is a stranger corner case than I thought! This code fails when the image is 600x400 but not 400x300

```cs
var plt = new ScottPlot.Plot(400, 300);

plt.PlotPoint(0, 1);
plt.PlotPoint(1, 1);
plt.PlotLine(0, 1.0000000000000009, 1, 1.0000000000000009);

string filename = System.IO.Path.GetFullPath("test.png");
plt.SaveFig(filename);
ProcessStartInfo psi = new ProcessStartInfo(filename) { UseShellExecute = true };
Process.Start(psi);
```

**Comment by swharden** [comment]
I can't believe this sucked up my whole day 😆 

It seems this error doesn't happen when you compile from source, but it does when you use the NuGet package. I had the hardest time figuring out why and I still haven't fully figured it out... but using the snupkg debug symbols I was able to determine that the `RecalculatePositionsAutomaticNumeric()` method in the tick mark generation system entered a near-infinite loop.

I'm pretty sure I fixed the problem with 6542c4f but I'm in the middle of a pretty big refactor right now (#578) so I probably won't push it to NuGet for a little while. The best way to use this fix is to run ScottPlot from source, but if your system is like mine then perhaps doing that would have fixed the problem all along!

Let me know if this doesn't do the trick 👍

**Comment by RFIsoft** [comment]
Scott,
I'm so sorry I caused this! I have already a work around, by rounding. So take your time! And it seems quite unique.
On the other hand, I think we both know from experience that something lingering needs to be squashed down, as it will likely cause issues down the road. Seems I had a real corner case.

Looks like added the applicable test indeed.

Thanks for the all the follow-up!
Best, Pete

**Comment by swharden** [comment]
> I'm so sorry I caused this!

No worries! I appreciate you bringing it to my attention

> something lingering needs to be squashed down, as it will likely cause issues down the road

Great quote! I suspect I fixed it with 6542c4f but I'll double-check it after I release the next NuGet package.

Best, 
Scott
